pep-0635.rst

```diff
@@ -0,0 +1,689 @@
+discerned by specific *values* (e.g., an object for 'addition' might be
+discerned by the ``operator`` field holding the string value ``'+'``),
+patterns can contain such values/constants.  The overall rules, however,
+specifically exclude actual expressions and make sure that only specific
+values are integrated into patterns.  The value ``-3``, for instance, is
+syntactically interpreted as the expression comprising the unary operation
+'negate' applied to the positive integer '3' (i.e. Python's syntax does
+not support negative numbers as atomic literals).  The overall syntax of
+patterns is carefully crafted to ensure that entities such as negative
+numbers can be included despite the exclusion of expressions in general.
+
+Nonetheless, it is desirable to express some constant values through named
+constants.  ``HttpStatus.OK``, for instance, might be much more readable
+than the plain number ``200``.  This poses a challenge, though, because
+the Python compiler cannot infer reliable from context, which names are
+meant to denote variables/parameters and which are meant to denote named
+constants.  Noting that many meaningful constants are organised in specific
+modules or enumerations, we follow a pragmatic approach here and interpret
+any dotted names as constants (recall that assignments to attributes are
+not possible because of side effects, anyway).  We acknowledge that this
+rule may seem restrictive as it leaves out support for named constants
+coming from the current namespace.  However, all alternatives turned out
+to either introduce much more complex rules or additional syntax.  We would
+also like to emphasise that better syntactic support for named constants
+could still be added in future proposals, thus warranting our focus on a
+minimal viable specficiation.
+
+
+
+.. _capture_pattern:
+
+Capture Patterns
+~~~~~~~~~~~~~~~~
+
+Capture patterns take on the form of a name that accepts any value and binds
+it to a (local) variable.  In that sense, a simple capture pattern is
+basically equivalent to a parameter in a function definition (when the
+function is called, eacg parameter binds the respective argument to a local
+variable in the function's scope).
+
+A name used for a capture pattern must not coincide with another capture
+pattern in the same pattern.  This, again, is similar to parameters, which
+equally require each parameter name to be unique within the list of
+parameters.  It differs, however, from iterable unpacking assignment, where
+the repeated use of a variable name as target is permissible (e.g.,
+``x, x = 1, 2``).  The rationale for not supporting ``(x, x)`` in patterns
+is its ambiguous reading: it could be seen as in iterable unpacking where
+only the second binding to ``x`` survives.  But it could be equally seen as
+expressing a tuple with two equal elements (which comes with its own issues).
+Should the need arise, then it is still possible to introduce support for
+repeated use of names later on.
```

Comment:
I added a paragraph on the store sigils (just below, lines 334-345).  But since the `^x` is already mentioned further below (with constant value patterns), I am not sure if it really adds much at this point.